docs/database/includes/azure-postgresql-ef-client.md

```diff
@@ -2,66 +2,263 @@
+
+```json
+{
+  "Aspire": {
+    "Npgsql": {
+      "EntityFrameworkCore": {
+        "PostgreSQL": {
+          "DisableHealthChecks": true,
+          "DisableTracing": true
+        }
+      }
+    }
+  }
+}
 ```
 
-### EF Core version 9+
+##### Use inline delegates
 
-With EF Core version 9, you can use the `ConfigureDataSource` method to configure the `NpgsqlDataSourceBuilder` that's used by the integration instead of building one outside of the integration and passing it in.
+You can configure settings in code, by passing the `Action<AzureNpgsqlEntityFrameworkCorePostgreSQLSettings> configureSettings` delegate to set up some or all the options inline, for example to disable health checks from code:
 
 ```csharp
-builder.AddNpgsqlDbContext<MyDb1Context>(
+builder.AddAzureNpgsqlDbContext<YourDbContext>(
     "postgresdb",
-    configureDbContextOptions: (options) => options.UseNpgsql(npgsqlOptions =>
-        npgsqlOptions.ConfigureDataSource(dsBuilder =>
-        {
-            if (string.IsNullOrEmpty(dsBuilder.ConnectionStringBuilder.Password))
-            {
-                var credentials = new DefaultAzureCredential();
-                var tokenRequest = new TokenRequestContext(["https://ossrdbms-aad.database.windows.net/.default"]);
-
-                dsBuilder.UsePasswordProvider(
-                    passwordProvider: _ => credentials.GetToken(tokenRequest).Token,
-                    passwordProviderAsync: async (_, ct) => (await credentials.GetTokenAsync(tokenRequest, ct)).Token);
-            }
-        })));
+    settings => settings.DisableHealthChecks = true);
+```
+
+Alternatively, you can use the `EnrichAzureNpgsqlDbContext` extension method to configure the settings:
+
+```csharp
+builder.EnrichAzureNpgsqlDbContext<YourDbContext>(
+    settings => settings.DisableHealthChecks = true);
+```
+
+<!-- TODO: Add xref to AzureNpgsqlEntityFrameworkCorePostgreSQLSettings.Credential when available -->
+
+Use the `AzureNpgsqlEntityFrameworkCorePostgreSQLSettings.Credential` property to establish a connection. If no credential is configured, the <xref:Azure.Identity.DefaultAzureCredential> is used.
+
+When the connection string contains a username and password, the credential is ignored.
+
+##### Troubleshooting
+
+In the rare case that the `Username` property isn't provided and the integration can't detect it using the application's Managed Identity, Npgsql throws an exception with a message similar to the following:
+
+> Npgsql.PostgresException (0x80004005): 28P01: password authentication failed for user ...
+
+In this case you can configure the `Username` property in the connection string and use `EnrichAzureNpgsqlDbContext`, passing the connection string in `UseNpgsql`:
+
+```csharp
+builder.Services.AddDbContextPool<YourDbContext>(
+    options => options.UseNpgsql(newConnectionString));
```

Comment:
I don't see anything in here configuring the user name.

Reply:
That's a good catch, how would one configure that?